pkg/aggregator/no_aggregation_stream_worker.go

```diff
@@ -135,13 +156,31 @@ func (w *noAggregationStreamWorker) run() {
 						n := time.Now()
 						if serializedSamples > 0 && lastStream.Before(n.Add(-time.Second*1)) {
 							log.Debug("noAggregationStreamWorker: triggering an automatic payloads flush to the forwarder (no traffic since 1s)")
+							tlmNoAggFlush.Add(1)
 							break mainloop // end `Serialize` call and trigger a flush to the forwarder
 						}
 
 					// receiving samples
 					case samples := <-w.samplesChan:
 						log.Debugf("Streaming %d metrics from the no-aggregation pipeline", len(samples))
+						var countProcessed int
+						var countUnsupportedType int
+
 						for _, sample := range samples {
+							mtype, supported := metricSampleAPIType(sample)
+
+							if !supported {
+								throttled, limit := w.logThrottling.ShouldThrottle()
+								if !throttled || limit {
+									log.Warnf("Discarding unsupported metric sample in the no-aggregation pipeline for sample '%s', sample type '%s'", sample.Name, sample.Mtype.String())
+								}
```

Comment:
I understand it's a lot more work but I think the log package should completely encapsulate this logic and the caller should not have to check if it should log or not each time.

Something like:
```
w.log = log.NewThrotthlingLogger(<some option>)
...
w.log.Errorf(...)
```

This should endup in the ASC roadmap but in the meantime I'm not sure adding the helper in `util` is worth it. This will make using the "final" throttling system much harder. Should we wait for the real implementation ? WDYT ?

Reply:
Moved it to `pkg/aggregator/internal/util` for now.